code/modules/events/tram_malfunction.dm

```diff
@@ -0,0 +1,66 @@
+#define TRAM_MALFUNCTION_TIME_UPPER 420
+#define TRAM_MALFUNCTION_TIME_LOWER 240
+#define XING_STATE_GREEN 0
+#define TRAM_ASSOCIATED_MAP "Tramstation"
+
+var/specific_lift_id = MAIN_STATION_TRAM
+var/original_lethality
+
+/datum/round_event_control/tram_malfunction
+	name = "Tram Malfunction"
+	typepath = /datum/round_event/tram_malfunction
+	weight = 20
+	max_occurrences = 4
+	category = EVENT_CATEGORY_ENGINEERING
+
+//If manually triggered, make sure we're actually on Tramstation. If not, cancel the event.
+/datum/round_event_control/tram_malfunction/admin_setup()
+	if(SSmapping.config.map_name == TRAM_ASSOCIATED_MAP)
+		return ..()
+
+	return ADMIN_CANCEL_EVENT
+
+/datum/round_event_control/tram_malfunction/can_spawn_event()
+	if(SSmapping.config.map_name == TRAM_ASSOCIATED_MAP)
```

Comment:
this is dumb

Reply:
look for length of tram signals list, check if theres any on station levels

Reply:
(and for the tram with the specific id you want to exist)

Reply:
Got it, now we look for MAIN_STATION_TRAM instead of a map check.